src/unit-test-coverage/shared/src/coveragetest-idmap.c

```diff
@@ -1065,6 +1070,76 @@ void Test_OS_ObjectIdIterator(void)
     UtAssert_STUB_COUNT(OS_Unlock_Global_Impl, 2);
 }
 
+void Test_OS_ObjectIDInteger(void)
+{
+    /*
+     * Test Case For:
+     * OS_ObjectIdToInteger, OS_ObjectIdFromInteger, OS_ObjectIdEqual, OS_ObjectIdDefined
+     */
+    int32             actual;
+    OS_object_token_t token;
+    osal_id_t         typesI[maxRecords];
+    osal_id_t         typesJ[maxRecords];
+    uint32            intID;
+    int32             recordscount = 0;
+    uint32            maxType      = 0;
+    uint32            typeReturn   = 0;
+    osal_objtype_t    idtype;
+    char              str[maxRecords];
+
+    for (idtype = 0; idtype < OS_OBJECT_TYPE_USER; ++idtype)
+    {
+        typeReturn = OS_GetMaxForObjectType(idtype);
+        maxType += typeReturn;
+    }
+
+    if (maxType == maxRecords)
+    {
+        for (idtype = 0; idtype < OS_OBJECT_TYPE_USER; ++idtype)
+        {
+            actual = OS_SUCCESS;
+            do
+            {
+                sprintf(str, "%d", recordscount);
+                actual = OS_ObjectIdAllocateNew(idtype, str, &token);
+
+                if (actual == OS_SUCCESS)
+                {
+                    typesI[recordscount] = token.obj_id;
+                    intID                = OS_ObjectIdToInteger(typesI[recordscount]);
+                    typesJ[recordscount] = OS_ObjectIdFromInteger(intID);
+
+                    recordscount++;
+                }
+
+            } while (actual == OS_SUCCESS);
+        }
+
+        for (int i = 0; i < recordscount; i++)
+        {
+            UtAssert_True(OS_ObjectIdDefined(typesI[i]), "%d Is defined", typesI[i]);
```

Comment:
Needs %lu format code and use "OS_ObjectIdToInteger()" to print.  Should not attempt to print osal_id_t directly - it might not be integer.

Reply:
fixed